x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/Expressions.java

```diff
@@ -121,10 +109,15 @@ public static AttributeSet references(List<? extends Expression> exps) {
             return AttributeSet.EMPTY;
         }
 
-        AttributeSet set = new AttributeSet();
+        int size = 0;
+        for (Expression exp : exps) {
+            size += exp.references().size();
```

Comment:
This re-computes the `references()` twice without storing them. This could introduce a lot more object creation than correctly sizing the Attribute set avoids. The AttributeSet computed in `Attribute.references()` is not stored for later retrieval.